buildkite/resource_pipeline_test.go

```diff
@@ -157,7 +157,7 @@ func TestAccPipeline_add_remove_withcluster_old_version(t *testing.T) {
 					resource.TestCheckResourceAttr("buildkite_pipeline.foobar", "cluster_id", "Q2x1c3Rlci0tLTFkNmIxOTg5LTJmYjctNDRlMC04MWYyLTAxYjIxNzQ4MTVkMg=="),
 					resource.TestCheckResourceAttr("buildkite_pipeline.foobar", "allow_rebuilds", "true"),
 					// check provider settings are not present
-					resource.TestCheckNoResourceAttr("buildkite_pipeline.foobar", "provider_settings"),
+					resource.TestCheckNoResourceAttr("buildkite_pipeline.foobar", "provider_settings.0"),
```

Comment:
Would this be better off checking the length of the list?

```go
TestCheckResourceAttr("buildkite_pipeline.foobar", "provider_settings.#", "0")
```

The above would check that the length is actually `0` and remove the risk that _something_ is set at `1`.

Ref: https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/teststep#builtin-check-functions